internal/transport/http2_server.go

```diff
@@ -333,7 +334,7 @@ func NewServerTransport(conn net.Conn, config *ServerConfig) (_ ServerTransport,
 		t.loopy.ssGoAwayHandler = t.outgoingGoAwayHandler
 		if err := t.loopy.run(); err != nil {
```

Comment:
Does loopy ever return with a nil error?

If not, let's always log (since we always log anyway)?  (I.e. remove the `if`.)